app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class RoomsController < ApplicationController
+  before_action :set_room, only: %i[show update]
+  # GET /rooms/:room_hash
+  def show
+    @jams = @room.jams.order('jams.created_at DESC').to_a
+    @current_jam = @jams.shift
+  end
+
+  # POST /rooms
+  def create
+    @room = Room.new
+
+    @room.save
+    redirect_to room_path(@room.room_hash), notice: 'Room was successfully created.'
+  end
+
+  # PATCH /rooms/:room_hash
+  def update
+    if params[:jam]
```

Comment:
that doesn't seem to be updating the room but creating a jam, so it either deserves its own nested resource (likely) or, at least, a proper action name.